src/optimizer.cpp

```diff
@@ -44,112 +47,181 @@ void Optimizer::getParams()
-  getParam(batch_size_, "batch_size", 200);
-  getParam(iteration_count_, "iteration_count", 2);
-  getParam(temperature_, "temperature", 0.25);
-
-  getParam(base_constraints_.vx, "vx_max", 0.5);
-  getParam(base_constraints_.vy, "vy_max", 1.3);
-  getParam(base_constraints_.wz, "wz_max", 1.3);
-  constraints_ = base_constraints_;
+  auto & s = settings_;
+  getParam(s.model_dt_, "model_dt", 0.1);
+  getParam(s.time_steps_, "time_steps", 12);
+  getParam(s.batch_size_, "batch_size", 400);
+  getParam(s.iteration_count_, "iteration_count", 2);
+  getParam(s.temperature_, "temperature", 0.25);
+  getParam(s.base_constraints_.vx, "vx_max", 0.5);
+  getParam(s.base_constraints_.vy, "vy_max", 0.5);
+  getParam(s.base_constraints_.wz, "wz_max", 1.3);
+  getParam(s.sampling_std_.vx, "vx_std", 0.2);
+  getParam(s.sampling_std_.vy, "vy_std", 0.2);
+  getParam(s.sampling_std_.wz, "wz_std", 1.0);
+  s.constraints_ = s.base_constraints_;
 
-  getParam(sampling_std_.vx, "vx_std", 0.1);
-  getParam(sampling_std_.vy, "vy_std", 0.1);
-  getParam(sampling_std_.wz, "wz_std", 0.3);
   getParentParam(controller_frequency_, "controller_frequency", 0.0);
 
   std::string motion_model_name;
   getParam(motion_model_name, "motion_model", std::string("DiffDrive"));
   setMotionModel(motion_model_name);
 }
 
+// FIXME(artofnothingness) optimize this
+rcl_interfaces::msg::SetParametersResult Optimizer::dynamicParametersCallback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+
+  rcl_interfaces::msg::SetParametersResult result;
+
+  auto getParamDouble =
+    [this](auto & setting, const std::string & param_name, const auto & parameter) {
+      const auto & type = parameter.get_type();
+      const auto & name = parameter.get_name();
+
+      if (type == rclcpp::ParameterType::PARAMETER_DOUBLE) {
+        if (name == param_name) {
+          auto param_value = parameter.as_double();
+          RCLCPP_INFO(logger_, "Parameter %s set to %f", param_name.c_str(), param_value);
+          setting = param_value;
+        }
+      }
+    };
+
+
+  auto getParamInt =
+    [this](auto & setting, const std::string & param_name, const auto & parameter) {
+      const auto & type = parameter.get_type();
+      const auto & name = parameter.get_name();
+
+      if (type == rclcpp::ParameterType::PARAMETER_INTEGER) {
+        if (name == param_name) {
+          auto param_value = parameter.as_int();
+          RCLCPP_INFO(logger_, "Parameter %s set to %ld", param_name.c_str(), param_value);
+          setting = param_value;
+        }
+      }
+    };
+
+  auto getParam = [&](auto & setting, const std::string & param_name, const auto & parameter) {
+      std::string param_full_name = name_ + "." + param_name;
+      getParamDouble(setting, param_full_name, parameter);
+      getParamInt(setting, param_full_name, parameter);
+    };
+
+  auto & s = settings_;
+  for (auto parameter : parameters) {
+    getParam(s.model_dt_, "model_dt", parameter);
```

Comment:
We know the types of each of these variables, we should just call `getParamDouble` or `getParamInt` as appropriate, its not like these types are changing (should you chose to do this lamda way of dynamic param getting, which I think is definitely reasonable)

Reply:
Then can remove the lambda `getParam`

Reply:
I just wanted the similar interface like in another getParam, sure it need some rework

Reply:
I reworked param handler, not relevant now